pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/ComputeStep.java

```diff
@@ -79,9 +81,13 @@ public void process(TransformContext transformContext) {
   }
 
   public void computeValueFields(List<ComputeField> fields, TransformContext context) {
-    if (context.getValueSchema().getSchemaInfo().getType() == AVRO) {
+    SchemaType schemaType = context.getValueSchema().getSchemaInfo().getType();
+    if (schemaType == AVRO) {
```

Comment:
nit:
what about pushing this into a method that accepts the schemaType ? This way we will have less point in the code of the steps that deal with AVRO/JSON explicitly, and it will be easier to maintain the code (and add new steps)
`context.addOrReplaceFields(schemaType, evaluatedFields, valueSchemaCache);`

Reply:
Good catch! done.